### PR TITLE
Allow custom bottomless db id for namespace

### PR DESCRIPTION
**Pull request by haaawk**

### CODE REVIEW

**Comment by haaawk** [comment]
Depends on https://github.com/tursodatabase/libsql/pull/564

File: libsql-server/src/connection/config.rs
Comment (athoscouto) [inline libsql-server/src/connection/config.rs:33]: Have you considered adding something like `bottomless_url` instead?
It would give us the extra flexibility of decoupling the bottomless URL from the namespace name.
And we could reuse the concept, for example, in the call to fork databases from a different `libsql-server` cluster.

Comment (haaawk) [reply libsql-server/src/connection/config.rs:33]: bottomless_db_id is really a bottomless_url. Namespace name/id is appended to it before usage here (https://github.com/tursodatabase/libsql/pull/596/files#diff-8fb5af38655652ae31867fe58be95481a3dcfba6bfa8ddda5f43fd39a9a4fea9R731). I think we can reuse it for fork already, no?

**Comment by haaawk** [comment]
This is ready for review. Tested with https://github.com/tursodatabase/turso/pull/1194

File: libsql-server/src/namespace/mod.rs
Comment (Horusiath) [inline libsql-server/src/namespace/mod.rs:177]: Option of option kinda feels like a red flag - its meaning is also ambiguous and hard to guess. Maybe make it a separate enum:

```rust
enum BottomlessIdInit {
  Explicit(String), // or something that describes your use case
  FromNamespace(String),
  NotProvided // nowadays it's pretty sure a configuration error and doesn't happen on prod
}
```

File: libsql-server/src/namespace/mod.rs
Comment (Horusiath) [inline libsql-server/src/namespace/mod.rs:739]: Same here: with dedicated enum, we could also pass one param instead of pair of `namespace_db_id`, `name`.

**Review by Horusiath — CHANGES_REQUESTED** [review]
One change proposal. Also maybe include the motivation/issue that this PR is supposed to cover. It will be easier to review given a context.

**Comment by psarna** [comment]
We also have precedence for using only env variables for bottomless config, maybe we can do the same here? To keep the interface free from bottomless-specific stuff

**Comment by MarinPostma** [comment]
@sarna this is a per-namespace configuration option, env is not gonna help. I do agree that this implementation is quite leaky, but I suppose we can roll with it for now, and revisit with unification.

I second @Horusiath comments

File: libsql-server/src/http/admin/mod.rs
Comment (LucioFranco) [inline libsql-server/src/http/admin/mod.rs:248]: I think we should make this fn accept a struct instead of parameters. This is going to conflict with the auth work I was doing. If possible I'd like to maybe hold off on merging this now and then I can merge my code then rebase this one ontop of it? Threading through stuff through this interface is a PITA so I can make that struct change then we can rebase this ontop to make things easier.

Comment (LucioFranco) [reply libsql-server/src/http/admin/mod.rs:248]: I've opened a draft PR with my latest work rebased, https://github.com/tursodatabase/libsql/pull/616 lets chat about this tomorrow.

File: libsql-server/src/namespace/mod.rs
Comment (LucioFranco) [reply libsql-server/src/namespace/mod.rs:177]: and the doc comments like that also make it really easy to understand 😄

File: libsql-server/src/namespace/mod.rs
Comment (LucioFranco) [inline libsql-server/src/namespace/mod.rs:191]: We want to avoid using context for now, as its designed to be used for cli apps and doesn't work well for our server reporting methods. The reason being that this only does the full chain print when its printed via debug not display but we use display. This will hide the internal error message and make it really hard to debug. I would instead defer to error logging at the site here the error like `tracing::error!("could not load database: {?}", err)` and then returning the error not wrapped in this context call.

**Review by LucioFranco — CHANGES_REQUESTED** [review]
Tomorrow lets chat about how we can merge this and the work I was already doing in https://github.com/tursodatabase/libsql/pull/616, maybe we want to start our own branch that we can merge into and then have one big rebase/merge into main again. That way we can have main as shippable.

File: libsql-server/src/namespace/mod.rs
Comment (haaawk) [reply libsql-server/src/namespace/mod.rs:177]: done.

File: libsql-server/src/namespace/mod.rs
Comment (haaawk) [reply libsql-server/src/namespace/mod.rs:739]: done.

File: libsql-server/src/namespace/mod.rs
Comment (haaawk) [reply libsql-server/src/namespace/mod.rs]: I was following the existing code but happy to improve. Done.

**Pull request merged**
